packages/rollup-plugin/src/rollupUtils.ts

```diff
@@ -46,10 +46,13 @@ function formatSourcemapData(
   // the same name as the js file... however we can pull the file name
   // from the sourcemap source just in case.
   let jsFilename: string
-  if (typeof sourcemap.source === 'string') {
+  if (sourcemap.source && typeof sourcemap.source === 'string') {
     const { file } = JSON.parse(sourcemap.source)
-    // The file in the source won't have the subfolder, need to add it
-    jsFilename = path.join(path.dirname(sourcemapFilename), file)
+    // The file in the source usually won't have the subfolder, we may need to add it
+    const subfolder = path.dirname(sourcemapFilename)
+    jsFilename = file.startsWith(subfolder) 
```

Comment:
Wouldn't this break if we have a file with the same name as its parent folder?
I'm wondering if it would be safer to break down `file` directory tree and compare paths with `subfolder`. What do you think?
This is the test I am referring to:
```
'bar/bar.js.map': {
    fileName: 'bar/bar.js.map',
    name: undefined,
    source: '{"version":3,"file":"bar.js","sources":["../../src/bar/bar.js"],"sourcesContent":["export default \'This is in a subfolder!\'"],"names":[],"mappings":";;AAAA,UAAe;;;;"}',
    type: 'asset' as const,
    needsCodeReference: false
  },
```

Reply:
Good point, will push an update. 

This whole section of code is kind of brittle because I haven't been able to find clear documentation from rollup on exactly what will be in the `file` field, nor have I seen documentation stating that the map will always have the same name as the file (though I've never seen a bundle come out otherwise). I'm not sure if pulling the `file` from the sourcemap is actually more reliable than just assuming the js file has the same name as the sourcemap, but it seems like slightly less of an assumption.